### ISSUE TITLE
Release or merge master into v0.9.0

### ISSUE THREAD

**Issue by dflemstr** [serde-rs/json#169]
I'm targeting the v0.9.0 branch which has been around for a long time, and it hasn't been merged with master nor released yet. When will that happen?

**Comment by dtolnay** [comment]
I pushed 3aa2f0fc03d77e0bc6645a80b39eed76e9cd0e45 which merges everything up through the 0.8.3 release into the v0.9.0 branch. I will plan to do the same thing at every future 0.8.x release.

As for a real 0.9 release, I am planning to do Serde 0.9 around the time that Macros 1.1 enters beta, which is hopefully in the next few months. That gives us a chance to resolve a handful of API glitches before the (hopefully) wave of adoption as it enters stable. Serde-json 0.9 will happen at the same time unless we find a compelling reason to do it earlier.

Right now as far as I know you are the only person with a stake in serde-json 0.9 so I don't think it makes sense to release just for this feature. Some libraries use `serde_json::Value` in public API so having semver-incompatible mismatches across libraries can cause trouble and I would prefer to do that as rarely as possible.

**Comment by dflemstr** [comment]
@dtolnay makes sense, but what do you mean by `serde_json::Value` changes?  There are no differences there between branches as far as I can see.

**Comment by dtolnay** [comment]
You can see the consequences now with crates that have not updated from serde_json 0.7 to 0.8. There were no changes to Value between 0.7 and 0.8. And yet:

```toml
[dependencies]
jamal = "0.1"
serde_json = "0.8"
```

```rust
extern crate jamal;
extern crate serde_json;

fn main() {
    jamal::to_yaml(&serde_json::Value::Null);
}
```

```
error[E0308]: mismatched types
 --> src/main.rs:5:20
  |
5 |     jamal::to_yaml(&serde_json::Value::Null);
  |                    ^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `serde_json::value::Value`, found enum `serde_json::Value`
  |
  = note: expected type `&serde_json::value::Value`
  = note:    found type `&serde_json::Value`
```

From Rust's perspective it is irrelevant whether `serde_json::Value` has changes, because 0.7 Value is a totally different type from 0.8 Value.

**Comment by dtolnay** [comment]
This is on track to be released with serde 0.9.0. We are tracking the release at https://github.com/serde-rs/serde/issues/668.
